pupil_src/shared_modules/marker_tracker_3d/controller.py

```diff
@@ -0,0 +1,26 @@
+from marker_tracker_3d import optimization
+from marker_tracker_3d.basic_models import CameraModel, MarkerModel
+from marker_tracker_3d.camera_localizer import CameraLocalizer
+from marker_tracker_3d.marker_detector import MarkerDetector
+
+
+class Controller:
+    def __init__(self, storage, update_menu):
+        self.storage = storage
+        self.storage.camera_model = CameraModel(cameraMatrix=None, distCoeffs=None)
+        self.storage.marker_model = MarkerModel()
+
+        self.marker_detector = MarkerDetector(self.storage)
+        self.optimization_controller = optimization.Controller(
+            self.storage, update_menu
+        )
+        self.camera_localizer = CameraLocalizer(self.storage)
+
+    def recent_events(self, frame):
+        self.marker_detector.detect(frame)
+
+        self.optimization_controller.fetch_extrinsics()
```

Comment:
By now I feel like `optimization` and `optimization_controller` are bad names. As a user of these things I am not really interested in if this is an optimization or some analytical computations or something else, but I want to know that this is my 3D Model. Maybe we can come up with a more descriptive name, something like `WorldModel` or `MarkerPoses` or something different.